api/v1/ptpconfig_webhook.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"errors"
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"strings"
+)
+
+// log is for logging in this package.
+var ptpconfiglog = logf.Log.WithName("ptpconfig-resource")
+
+func (r *PtpConfig) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+type ptp4lConfSection struct {
+	options map[string]string
+}
+
+type ptp4lConf struct {
+	sections map[string]ptp4lConfSection
+}
+
+func (output *ptp4lConf) populatePtp4lConf(config *string, ptp4lopts *string) error {
+	lines := strings.Split(*config, "\n")
+	var currentSection string
+	output.sections = make(map[string]ptp4lConfSection)
+
+	for _, line := range lines {
+		if strings.HasPrefix(line, "[") {
+			currentSection = line
+			currentLine := strings.Split(line, "]")
+
+			if len(currentLine) < 2 {
+				return errors.New("Section missing closing ']'")
+			}
+
+			currentSection = fmt.Sprintf("%s]", currentLine[0])
+			section := ptp4lConfSection{options: map[string]string{}}
+			output.sections[currentSection] = section
+		} else if currentSection != "" {
+			split := strings.IndexByte(line, ' ')
+			if split > 0 {
+				section := output.sections[currentSection]
+				section.options[line[:split]] = line[split+1:]
+				output.sections[currentSection] = section
+			}
+		} else {
+			return errors.New("Config option not in section")
+		}
+	}
+	_, exist := output.sections["[global]"]
+	if !exist {
+		output.sections["[global]"] = ptp4lConfSection{options: map[string]string{}}
+	}
+
+	// When validating, add ptp4lopts to conf for fields we check
+	opts := strings.Split(*ptp4lopts, " ")
+	for index, opt := range opts {
+		if opt == "--summary_interval" && index < len(opts)-1 {
+			output.sections["[global]"].options["summary_interval"] = opts[index+1]
+		}
+	}
+	return nil
+}
+
+func (r *PtpConfig) validate() error {
+	profiles := r.Spec.Profile
+	for _, profile := range profiles {
+		conf := &ptp4lConf{}
+		conf.populatePtp4lConf(profile.Ptp4lConf, profile.Ptp4lOpts)
```

Comment:
the problem here is that you don't load also the defaults from the /etc/ptp.conf file, so you will not have all the configuration
you may need to install linuxptp package also in the operator to have access to the file